jctools-core/src/main/java/org/jctools/queues/BaseMpscLinkedArrayQueue.java

```diff
@@ -403,7 +417,6 @@ else if (casProducerIndex(pIndex, pIndex + 1))
     {
         final long offset = nextArrayOffset(mask);
         final E[] nextBuffer = (E[]) lvElement(buffer, offset);
-        soElement(buffer, offset, null);
```

Comment:
This will cause GC nepotism

Reply:
Thanks, wasn't aware of this issue. Will try to come up with a solution

Reply:
I replaced this with a marker object that will cause the iterator to jump to the current consumerBuffer.